src/datreant/core/tests/test_trees.py

```diff
@@ -221,6 +221,36 @@ def test_glob(self, tree):
         assert tl in tree.glob('*r*y')
         assert tc in tree.glob('*r*y')
 
+    def test_walk(self, tree, tmpdir):
+        with tmpdir.as_cwd():
+            tree['scipy'].make()
```

Comment:
All three of these items will be files. We also need to make some directories, preferably with some files inside them. At the moment without any directories the directory checks at the bottom will see empty lists.